.github/workflows/github-actions-ci.yml

```diff
@@ -22,30 +21,34 @@ jobs:
           echo "List of changed files:"
           echo ${CHANGED_PYFILES}
           flake8 ${CHANGED_PYFILES} --count
-
+  
   ecephys_etl_pipelines:
-    name: ${{ matrix.python-version }}, ${{ matrix.os }}
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: ["macos-latest", "windows-latest", "ubuntu-latest"]
-        python-version: ["3.6", "3.7"]
-    defaults:
-      run:
-        shell: bash -l {0}
+    runs-on: "ubuntu-latest"
     steps:
-      - uses: actions/checkout@v2
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
         with:
-          auto-update-conda: true
-          python-version: ${{ matrix.python-version }}
-          activate-environment: test-env
-      - name: Install
+          username: alleninstitutepika
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Docker Image
         run: |
-          conda activate test-env
-          pip install codecov
-          pip install -r test_requirements.txt
-          pip install .
-      - name: Test
+          git clone https://github.com/AllenInstitute/ecephys_etl_pipelines.git
+          cd ecephys_etl_pipelines
+          echo "Using branch $GITHUB_HEAD_REF"
+          git checkout $GITHUB_HEAD_REF
+          docker build --build-arg ECEPHYS_ETL_TAG=${GITHUB_HEAD_REF} --build-arg ECEPHYS_ETL_COMMIT_SHA=${GITHUB_SHA} --tag alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA docker/
+      - name: Run Tests in Docker Image
         run: |
-          py.test --cov=ecephys_etl_pipelines -n 4
+          docker run alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA pytest .
+      - name: Push Docker Image
+        run: |
+            if [[ "$GITHUB_HEAD_REF" == "main" ]]
+            then
+                docker tag alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA alleninstitutepika/ecephys_etl_pipelines:main
+                docker push alleninstitutepika/ecephys_etl_pipelines:main
+            else
+                docker tag alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA alleninstitutepika/ecephys_etl_pipelines:develop
+                docker push alleninstitutepika/ecephys_etl_pipelines:develop
+            fi
+
+            docker push alleninstitutepika/ecephys_etl_pipelines:$GITHUB_SHA
```

Comment:
you are pushing everything 2x.

Reply:
Are you sure? One should just be pushing the `develop` or `main` image, then the final one pushes it again but tagged with the hash. This looks to be what the `ophys_etl_pipeline` one does as well

Reply:
yeah, your 2nd push just pushes the 2nd tag. But it is not necessary. I see 1 push here:
https://github.com/AllenInstitute/ophys_etl_pipelines/blob/b73f30e044a39a00d075b297fa858eece14242b5/.circleci/config.yml#L33-L43

Reply:
Yeah I saw that too, but the same code caused a few errors here.
1. It would use the 'latest' tag for the push, but that hadn't been tagged so it would throw an error. Now I could just tag it latest instead of main or develop, but:
2. In the docker hub page for alleninstitutepika/ophys_etl_pipelines, it doesn't seem to use the 'latest' tag. It looks like it is very much having both the hash version AND the develop versions pushed regularly. So even though what you said is what it looked like the yml file was saying to do, it was very much unclear what was supposed to happen.

Reply:
Ok I did that. I'm not totally sure how it's different than what was there, other than being tidier. I'm also not sure how the `ophys` repo is uploading both without a `-a` flag... But it works now